pkg/danm/danm.go

```diff
@@ -183,72 +189,98 @@ func extractConnections(args *cniArgs) error {
   return nil
 }
 
-func setupNetworking(args *cniArgs) (*current.Result, error) {
-  syncher := syncher.NewSyncher(len(args.interfaces))
-  for nicId, nicParams := range args.interfaces {
-    nicParams.DefaultIfaceName = "eth" + strconv.Itoa(nicId)
-    go createInterface(syncher, nicParams, args)
+func getAllocatedDevices(args *cniArgs, devicePool string)(map[string][]string, error){
+  checkpoint, err := checkpoint_utils.GetCheckpoint()
+  if err != nil {
+    return nil, errors.New("failed to instantiate checkpoint object due to:" + err.Error())
   }
-  err := syncher.GetAggregatedResult()
-  return syncher.MergeCniResults(), err
+  resourceMap, err := checkpoint.GetComputeDeviceMap(string(args.podUid))
+  if err != nil || len(resourceMap) == 0 {
+    return nil, errors.New("failed to retrieve Pod info from checkpoint object due to:" + err.Error())
+  }
+  allocatedDevices := make(map[string][]string)
+  for resourcename, resources := range resourceMap {
+    if strings.HasPrefix(resourcename, devicePool) {
+      hostDevice, err := sriov_utils.GetPfName(resources.DeviceIDs[0])
```

Comment:
code is still SR-IOV specific here
it is also a stretch to think that every networking related device in the future will be tied to a NIC
why not simply save the DeviceIDs to allocatedDevices[resourceName]?

Reply:
better yet: just return a []string slice with the DeviceIDs belonging to one devicename

Reply:
Applied